### PR TITLE
Add /extensions endpoint

### PR DESCRIPTION
**Pull request by bbondy**
Still a WIP while I get this integrated in desktop, it might need more changes. Just throwing it up for now though.

### CODE REVIEW

**Comment by bbondy** [comment]
@aekeus this is ready for review, pls also deploy if you are ok with the changes.

File: src/controllers/extensions.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+          .up()
+        .up()
+      .up()
+  })
+  return doc.toString({ pretty: true })
+}
+
+const setup = (runtime, availableExtensions) => {
+  let extensionsRoute = {
+    method: ['POST'],
```
Comment (aekeus) [inline src/controllers/extensions.js]: Is a POST here a requirement of the extensions support? Generally a POST will mutate the state of the server.

File: src/controllers/extensions.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    path: '/extensions',
+    config: {
+      handler: function (request, reply) {
+        const requestedExtensions = getRequestedExtensions(request.payload.toString())
+        const extensionsWithUpdates = getExtensionsWithUpdates(availableExtensions, requestedExtensions)
+        reply(getExtensionsResponse('https://s3.amazonaws.com/brave-extensions/release', extensionsWithUpdates))
```
Comment (aekeus) [inline src/controllers/extensions.js]: Perhaps we can we move this to an environment variable?

File: src/controllers/extensions.js (hunk 1 above)
Comment (bbondy) [inline src/controllers/extensions.js]: yes chrome uses POST so it's just implementing what they use.

**Pull request merged**